yw_website/yw_website/apps/website/utils.py

```diff
@@ -43,3 +45,37 @@ def search_and_create_query_set(q):
         return Workflow.objects.filter(query).exclude(version__isnull=True)
     else:
         return Workflow.objects.none()
+
+def get_block_inputs(run_id):
+    '''
+    This function grabs inputs into the program blocks of a run.
+    KIDDING NO IT DOESN'T
+    This function sets up the block structure so that from a parent program
+    block you may drill down into a child of that program block
+    '''
+    program_blocks = ProgramBlock.objects.filter(run=run_id)
+    x = []
+    parent = ProgramBlock()
+    for block in program_blocks:
+        next_block = ProgramBlock()
+        if block.in_program_block == None:
+            parent.name = block.name
+            parent.programblock_id = block.programblock_id
+            parent.id = block.id
+            x.append(parent)
+            print(parent.name)
+            print(parent)
+            print()
+            
+        else:
+            next_block.name = block.name
+            next_block.programblock_id = block.programblock_id
```

Comment:
you can find direct descendants by filtering on the `program_blocks` queryset again on `in_program_block=next_block.programblock_id`